src/controllers/imap.js

```diff
@@ -1,127 +1,168 @@
-var Imap = require("imap");
+import Imap from "imap";
 import BaseAPIController from "./BaseAPIController";
 import ImapProvider from "../providers/ImapProvider";
 
 export class ImapController extends BaseAPIController {
 
-/* Controller for Save Imap Data  */
-	save = (req, res) => {
-		ImapProvider.save(this._db.Imap, req.checkBody, req.body, req.getValidationResult())
-.then((data) => {
-	this._db.Imap.create(data)
-.then(res.json.bind(res))
-.catch(this.handleErrorResponse.bind(null, res));
-})
-                       .catch(this.handleErrorResponse.bind(null, res));
-	}
+    /* Controller for Save Imap Data  */
+    save = (req, res) => {
+        ImapProvider.save(this._db.Imap, req.checkBody, req.body, req.getValidationResult())
+            .then((data) => {
+                this._db.Imap.create(data)
+                    .then(res.json.bind(res))
+                    .catch(this.handleErrorResponse.bind(null, res));
+            })
+            .catch(this.handleErrorResponse.bind(null, res));
+    }
```

Comment:
general changes to make in all controllers.

1. there should be only one catch in entire controller. Use throw new Error in other places
2. the idResult function should be last function in all controllers.
3. controller should be very very slim. very less code. all other code should be defined in model functions and services. if controller exceeds 5-6 lines of code there is some problem in code
4. no direct db operation in controller, even for create, update, destory. make mode functino instead of write it directly in controller.
5. don't hard code limit in controllers, always take from route. 
6. in model use boolean never   status: "FALSE" or statue: "TRue" this is wrong.